src/lib/resolvers/partialDMChannel.ts

```diff
@@ -0,0 +1,15 @@
+import { isDMChannel } from '@sapphire/discord.js-utilities';
+import type { Message, PartialDMChannel } from 'discord.js';
+import { Identifiers } from '../errors/Identifiers';
+import { err, ok, Result } from '../parsers/Result';
+import { resolveChannel } from './channel';
+
+export function resolvePartialDMChannel(
+	parameter: string,
+	message: Message
+): Result<PartialDMChannel, Identifiers.ArgumentChannelError | Identifiers.ArgumentDMChannelError> {
+	const result = resolveChannel(parameter, message);
+	if (!result.success) return result;
+	if (isDMChannel(result.value) && result.value.partial) return ok(result.value);
```

Comment:
I don't think we should discriminate the partiality, if anything, a non-partial channel already has all the properties partial channels have.